### PR TITLE
[sled agent] Deprecate the old service format

### PR DESCRIPTION
**Pull request by smklein**
This portion of the Sled Agent was added in https://github.com/oxidecomputer/omicron/pull/4466, migrating an old "services" ledger into a new, zones-oriented format.

This code served as a bridge between old and new format, but maintaining backwards compatibility here is a burden
for modifying the zone format further.

Specifically: in https://github.com/oxidecomputer/omicron/pull/5050 , which attempts to add the notion of "transient zone filesystem zpool" to the request from Nexus, the translation of "old format -> new format" is no longer possible without additional information about "where should the transient zone filesystem be placed".

Given that this conversion has been deployed for multiple months - across releases - I figured it would be easier to remove the old support, rather than making another bridge for old code compatibility.

### CODE REVIEW

File: sled-agent/src/services_migration.rs [hunk 1]
```diff
@@ -1,623 +0,0 @@
...
-            "rack3-sled22.json",
-            "rack3-sled23.json",
-            "rack3-sled24.json",
-            "rack3-sled25.json",
-            "rack3-sled26.json",
-            "rack3-sled27.json",
-            "rack3-sled28.json",
-            "rack3-sled29.json",
-            "rack3-sled30.json",
-            "rack3-sled31.json",
```
Comment (davepacheco) [inline sled-agent/src/services_migration.rs:604]: I think we can probably remove all of these files, too -- basically all of `omicron/sled-agent/tests/old-service-ledgers` and `omicron/sled-agent/tests/output/new-service-ledgers`.

File: sled-agent/src/services_migration.rs [hunk 2]
```diff
@@ -1,623 +0,0 @@
...
-
-    /// Verifies that our understanding of this old-format ledger has not
-    /// changed.  (If you need to change this for some reason, you must figure
-    /// out how that affects systems with old-format ledgers and update this
-    /// test accordingly.)
-    #[test]
-    fn test_all_services_requests_schema() {
-        let schema = schemars::schema_for!(AllZoneRequests);
-        expectorate::assert_contents(
-            "../schema/all-zone-requests.json",
```
Comment (davepacheco) [inline sled-agent/src/services_migration.rs:544]: Can we remove this file too?

File: sled-agent/src/services.rs [hunk 3]
```diff
@@ -815,127 +796,8 @@ impl ServiceManager {
             return Ok(Some(ledger));
         }
 
-        // Now look for the ledger used by previous versions.  If we find it,
```
Comment (davepacheco) [inline sled-agent/src/services.rs:818]: I kind of feel like we should try to remove any old-format ledger that we find here.  The discussion at L894-L936 is a good summary of the tradeoffs, except that now we know we're in the case where this ledger can't possibly be useful.  It also almost certainly can't be dangerous, either -- just potentially confusing to humans.

**Review by davepacheco — COMMENTED** [review]
🥳

File: sled-agent/src/services_migration.rs (hunk 1 above)
Comment (smklein) [reply sled-agent/src/services_migration.rs:604]: Good catch, definitely should have removed these -- done!

File: sled-agent/src/services_migration.rs (hunk 2 above)
Comment (smklein) [reply sled-agent/src/services_migration.rs:544]: Done

File: sled-agent/src/services.rs [hunk 4]
```diff
@@ -778,26 +794,45 @@ impl ServiceManager {
...
+        // cleanup.
+        for path in self.all_service_ledgers().await {
+            match tokio::fs::remove_file(&path).await {
+                Ok(_) => (),
+                Err(ref e) if e.kind() == std::io::ErrorKind::NotFound => (),
+                Err(e) => {
+                    return Err(Error::io(
+                        "Failed to delete old service ledger",
+                        e,
+                    ))
```
Comment (davepacheco) [inline sled-agent/src/services.rs:812]: I'd just `warn!` here and proceed, since we don't actually need it to be cleaned up.

Comment (smklein) [reply sled-agent/src/services.rs]: Done: a04e9c79f3aefba7f18be982fee09986a1c5d3bf

**Review by davepacheco — APPROVED** [review]

**Pull request merged**